### ISSUE TITLE
Idea: access to original chunks

### ISSUE THREAD

**Issue by douglasr** [joeyates/metar-parser#47]
I have a situation where I want to parse the METAR, storing various data in a database for querying but then also rendering the original chunks, as posted in the METAR. As an example, if the wind chunk was "36007KT", that is nicely parsed and stored as 0 degrees and 3.60111 m/s which I store in the database. However, I would like to be able to then render the original text to the user. Although I could reconstruct the output by converting units and handling any special cases (such as 0 deg being displayed as 360 deg) I'm thinking it might make more sense to simply have an internal hash which maintains the original text as parsed. @joeyates, is this something that you would accept a pull request for?

**Comment by joeyates** [comment]
This looks like a change which would **mostly** affect the data storage classes, and not the parser itself. I.e. as most of these classes receive raw chunks, all they would have to do is keep a copy of them.

Overall, such a change might benefit the codebase as logic would be taken out the parser and placed in the data storage classes, making them less than dumb containers and reducing coupling between them and the parser,

So, cases where code does as follows would not need to change:
```ruby
foo = Foo.parse(chunks[0])
if foo
  ...
```

Cases like 'CAVOK' on the other hand are problematic.

With 'CAVOK', one chunk of input affects 3 states: visibility, present weather and sky conditions (The same is true of "temperature and dew point", which affects two states).

How would you propose handling these cases?

As far as I can see the possibilities for 'CAVOK' are:
1. merge visibility, present weather and sky conditions together as a new, single, data store,
2. have some special flag in the parser for "CAVOK".

Both of these seem unsatisfactory.

Let me know what you think.

**Comment by douglasr** [comment]
I was simply thinking of keeping a hash with all of the various chunks, as encountered and without any transformation.

```ruby
{
  :time => '271844Z',
  :wind => '36007KT',
  :visibility => '1SM',
  :weather => '-RA BR',
  :sky => 'BKN015 OVC025',
  :temperature => '06/04',
  :altimeter => 'A2992',
  :rml => 'A01 DENSITY ALT 3400FT'
}
```

In the case of CAVOK it could probably be used for the :sky key only or used for both :sky and :visibility to reduce any confusion.

```ruby
{
  :visibility => 'CAVOK',
  :sky => 'CAVOK',
  :weather => nil,
}

I'm in favour of a single use of 'CAVOK' since that would allow you to reconstruct the METAR simply by concatenating the elements together in the proper order, ignoring any nil elements.

```ruby
{
  :visibility => nil,
  :sky => 'CAVOK',
  :weather => nil,
}

**Comment by joeyates** [comment]
I've released v1.3.0 with `Parser#raw_attributes`. It returns a `Hash` of the raw parsed data which you can then encode with `#to_json`
